evalml/tests/component_tests/test_delayed_features_transformer.py

```diff
@@ -272,3 +280,32 @@ def test_delay_feature_transformer_y_is_none(delayed_features_data):
                            "feature_delay_1": X.feature.shift(1),
                            })
     assert_frame_equal(answer, DelayedFeatureTransformer(max_delay=1, gap=11).fit_transform(X, y=None).to_dataframe())
+
+
+@pytest.mark.parametrize("X_df", [pd.DataFrame(pd.to_datetime(['20190902', '20200519', '20190607'], format='%Y%m%d')),
+                                  pd.DataFrame(pd.Series([1, 2, 3], dtype="Int64")),
+                                  pd.DataFrame(pd.Series([1., 2., 3.], dtype="float")),
+                                  pd.DataFrame(pd.Series(['a', 'b', 'a'], dtype="category")),
+                                  pd.DataFrame(pd.Series(['this will be a natural language column because length', 'yay', 'hay'], dtype="string"))])
+def test_delay_feature_transformer_woodwork_custom_overrides_returned_by_components(X_df):
+    y = pd.Series([1, 2, 1])
+    override_types = [Integer, Double, Categorical, Datetime, Boolean]
+    for logical_type in override_types:
+        try:
+            X = ww.DataTable(X_df, logical_types={0: logical_type})
+        except TypeError:
+            continue
+        dft = DelayedFeatureTransformer(max_delay=1, gap=11)
+        dft.fit(X, y)
+        transformed = dft.transform(X, y)
+        assert isinstance(transformed, ww.DataTable)
+        if logical_type == Integer or logical_type == Double or logical_type == Categorical:
+            assert transformed.logical_types == {0: logical_type,
```

Comment:
I think we should modify the delayed feature transformer to carry over the ints. I'd be happy to file an issue if you are on board!

Reply:
Sure, that sounds good! I'm guessing `shift` just returns floats which is why we end up with Doubles right now?

Reply:
Yes!